presto-main/src/main/java/com/facebook/presto/sql/planner/optimizations/PruneUnreferencedOutputs.java

```diff
@@ -723,11 +723,6 @@ public PlanNode visitValues(ValuesNode node, RewriteContext<Set<Symbol>> context
         @Override
         public PlanNode visitApply(ApplyNode node, RewriteContext<Set<Symbol>> context)
         {
-            // remove unused apply nodes
-            if (intersection(node.getSubqueryAssignments().getSymbols(), context.get()).isEmpty()) {
-                return context.rewrite(node.getInput(), context.get());
-            }
-
```

Comment:
I don't think it makes sense to remove this until we've migrated all of the pruning cases -- the other pruning cases could cause this code to run.